### ISSUE TITLE
RegisterModule doesn't generate with the corresponding generic.

### ISSUE THREAD

**Issue by Darkness4** [Milad-Akarie/injectable#19]
Hello,

I'm using `injectable: 0.2.3`.
When using generic types, `RegisterModule` is not able to register an instance with the corresponding generic.

For example :

```dart
@injectable
class Example {
  final ThirdParty<String> thirdParty;

  Example(this.thirdParty);
}

class ThirdParty<T> {}

@registerModule
abstract class RegisterModule {
  ThirdParty<String> get thirdParty;
}
```

This generates :

```dart
void $initGetIt(GetIt g, {String environment}) {
  final registerModule = _$RegisterModule();
  g.registerFactory<ThirdParty>(() => registerModule.thirdParty);
  g.registerFactory<Example>(() => Example(
        g<ThirdParty<String>>(),
      ));
}

class _$RegisterModule extends RegisterModule {
   // '_$RegisterModule.thirdParty' ('ThirdParty<dynamic> Function()') isn't a 
  // valid override of 'RegisterModule.thirdParty' ('ThirdParty<String> Function()').dart
  // (invalid_override)
  ThirdParty get thirdParty => ThirdParty();
}
```

And i get an error from `build_runner`:

```txt
[Example] depends on [ThirdParty<String>] which is not injectable!
```

**Comment by rubgithub** [comment]
Try to add @injectable in

```
@registerModule
@injectable
abstract class RegisterModule
```

**Comment by Darkness4** [comment]
> Try to add @Injectable in
> 
> ```
> @registerModule
> @injectable
> abstract class RegisterModule
> ```

There's no change in behavior (same file generated, same errors).
I'm trying to register a third party instance which is a generic type.

More specifically, i'm trying to register a [Hive](https://pub.dev/packages/hive) box :

```dart
import 'package:get_it/get_it.dart';
import 'package:hive/hive.dart';
import 'package:injectable/injectable.dart';
import 'injection_container.iconfig.dart';

final sl = GetIt.instance;

@injectableInit
void init({String env = Environment.prod}) => $initGetIt(sl, environment: env);

@registerModule
abstract class RegisterModule {
  @prod
  Box<String> get box => Hive.box<String>('prefs');
}
```

```dart
import 'package:flutter/foundation.dart';
import 'package:hive/hive.dart';
import 'package:injectable/injectable.dart';

abstract class Service {}

@RegisterAs(Service)
@prod
@lazySingleton
@injectable
class ServiceImpl implements Service {
  final Box<String> box;

  ServiceImpl({@required this.box});
}
```

Which generates : 

```dart
void $initGetIt(GetIt g, {String environment}) {
  final registerModule = _$RegisterModule();

  if (environment == 'prod') {
    g.registerLazySingleton<Service>(() => ServiceImpl(
          box: g<Box<String>>(),
        ));
    g.registerFactory<Box>(() => registerModule.box);  // Was expecting <Box<String>>
  }
}

class _$RegisterModule extends RegisterModule {}
```

**Comment by Milad-Akarie** [comment]
@Darkness4 should be fixed in injectable_generator: 0.2.4 check it out!
